app/helpers/search_helper.rb

```diff
@@ -1,12 +1,12 @@
 module SearchHelper
-  def check_posts_permissions
-    (current_user&.is_moderator || current_user&.is_admin ? Post : Post.undeleted)
+  # @param user [User] user to check
+  def accessible_posts_for(user)
+    (user&.is_moderator || user&.is_admin ? Post : Post.undeleted)
       .qa_only.list_includes
   end
```

Comment:
Should this also check category access settings here? We've got an open security report that posts in private categories sometimes show in search; this might be an ideal place to check that.

Reply:
Yup, looks like just the place to do so - will update once I am done with basic test coverage for the helper (want to make sure there are no other surprises first)

Reply:
Or actually sooner than that as properly testing the `:category` qualifier requires the check to work correctly.

Reply:
After reading through the [security report](https://github.com/codidact/qpixel/security/advisories/GHSA-pv74-hcg9-65r4) itself, I realized that it doesn't seem to be related to search, @ArtOfCode-, am I missing something? I abstracted the check for category access into `accessible_categories_for` just in case, but the tests show that search handles category access correctly.

Reply:
That's a different report - there's another one open in the support tickets that hasn't made it to a GH advisory yet.

Reply:
Ah, got it, there is another one. Hmm, looks like that one might crop up when there is no filter by category. Will take a look

Reply:
Deferring the second vulnerability fix (for the publicly reported one) to another PR - this now takes us over the finish line with switching pipelines, so it's best not to stall